src/haxeLanguageServer/features/haxe/codeAction/DiagnosticsCodeActionFeature.hx

```diff
@@ -203,9 +210,164 @@ class DiagnosticsCodeActionFeature implements CodeActionContributor {
 			});
 		}
 
+		final tooManyArgsRe = ~/Too many arguments([\w.]*)/;
+		if (tooManyArgsRe.match(arg)) {
+			final document = context.documents.getHaxe(params.textDocument.uri);
+			final replacement = document.getText(diagnostic.range);
+			actions.push({
+				title: "Add argument",
+				data: CodeActionFeature.addResolveData(MissingArg(action -> createMissingArgumentsAction(action, params, diagnostic))),
+				kind: QuickFix,
+				diagnostics: [diagnostic],
+				isPreferred: true,
+			});
+		}
 		return actions;
 	}
 
+	function createMissingArgumentsAction(action:CodeAction, params:CodeActionParams, diagnostic:Diagnostic):Null<Promise<CodeAction>> {
+		final document = context.documents.getHaxe(params.textDocument.uri);
+		if (document == null)
+			return null;
+		var fileName:String = document.uri.toFsPath().toString();
+		final pos = document.offsetAt(diagnostic.range.start);
+		var tokenSource = new CancellationTokenSource();
+		final gotoDefinition = new GotoDefinitionFeature(context, false);
```

Comment:
`Context` has a field `gotoDefinition` which should provide access to `onGotoDefinition`, so a private instance shouldn't be needed.